### ISSUE TITLE
Complete switch from SQLite to Firebase

### ISSUE THREAD

**Issue by redlor** [redlor/Minitask#4]

**Comment by redlor** [comment]
I completed this in the Firebase branch, not merged yet because needs testing on many possible scenarios.

**Comment by danielblokus** [comment]
@redlor app displays the toast with error 'Developer error' after successfully sign in via Google 
![screenshot_1514459097](https://user-images.githubusercontent.com/9052896/34409016-76acf996-ebc7-11e7-8e24-49eed7b1ac9b.png)

**Comment by danielblokus** [comment]
@redlor There are few things:

- also you should delete Database package with classes: TodoListContract, TodoListDbHelper; <- unused code should be deleted

- use a Logger  instead System.out.println to print debug info

- duplicated code in MainActivity class, line: 80. Maybe call the onSignedOutCleanup() method from line: 161 method

**Comment by danielblokus** [comment]
@redlor duplicated code in MainActivity class, line: 80. Maybe call onSignedOutCleanup() from line: 161 method

**Comment by redlor** [comment]
@danielblokus 
I think the developer error happens because I did not insert in a release fingerprint key, but just the developer one. that means I am the only one who can log in. This should happen only with Google Sign in, can you please try with email + plus password sign in and let me know if it works?

- TodoListContract, TodoListDbHelper: I cleaned all unused code, just forgot these ones, thank you for spotting.

- use a Logger instead System.out.println: I know, but I am old style....Anyway I will clear them when i won't need anymore

- duplicated code in MainActivity class, line: 80: done.

I found a bug in the 7 days fragment, it is showing old items too, working on a fix. 
In addition, I am almost done with the data persistence offline, I will soon make another push.

thank you so much for your help, very much appreciated!!!

**Comment by danielblokus** [comment]
@redlor 
Yeah, sign in via email works well. You mentioned offline... Is it possible to bypass the firebase database and go offline while running UI tests? If I know how to set firebase to offline, I would prepare some UI tests for this branch.

**Comment by redlor** [comment]
@danielblokus 
I see: 
![image](https://user-images.githubusercontent.com/25771097/34421034-3cac3720-ec0d-11e7-8550-f5b966c09421.png)

btw I just pushed offline capabilities if you want to test them. I am not sure about bypassing the database for testing, couldn't find anything in the official documentation.
